ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/gutter/Gutter.java

```diff
@@ -17,47 +17,54 @@
   /**
    * Adds a gutter item.
    *
-   * @param lineStart the first line of the item
-   * @param lineEnd the last line of the item
+   * @param line the line for the item
    * @param gutterId the gutter identifier
    * @param element the (DOM) element to add
    */
-  void addGutterItem(int lineStart, int lineEnd, String gutterId, Element element);
+  void addGutterItem(int line, String gutterId, Element element);
 
   /**
    * Adds a gutter item.
    *
+   * <p>If is not overridden, then equals to {@link #addGutterItem(int, String, Element)} with
+   * <code>lineStart</code> as an argument, <code>lineEnd</code> will be ignored.
+   *
    * @param lineStart the first line of the item
    * @param lineEnd the last line of the item
    * @param gutterId the gutter identifier
    * @param element the (DOM) element to add
+   */
+  default void addGutterItem(int lineStart, int lineEnd, String gutterId, Element element) {
+    addGutterItem(lineStart, gutterId, element);
```

Comment:
But this will add several single line gutter items. I think this is kind of hidden functionality.